service/controller/resource/alert/resource.go

```diff
@@ -35,66 +44,61 @@ func New(config Config) (*generic.Resource, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Installation must not be empty", config)
 	}
 
-	clientFunc := func(namespace string) generic.Interface {
-		c := config.PrometheusClient.MonitoringV1().PrometheusRules(namespace)
-		return wrappedClient{client: c}
-	}
-
-	c := generic.Config{
-		ClientFunc:    clientFunc,
-		Logger:        config.Logger,
-		Name:          Name,
-		GetObjectMeta: getObjectMeta,
-		GetDesiredObject: func(obj interface{}) (metav1.Object, error) {
-			return getRules(obj, config.Installation)
-		},
-		HasChangedFunc: hasChanged,
-	}
-	r, err := generic.New(c)
-	if err != nil {
-		return nil, microerror.Mask(err)
+	r := &Resource{
+		logger:           config.Logger,
+		prometheusClient: config.PrometheusClient,
+		installation:     config.Installation,
 	}
 
 	return r, nil
 }
 
-func getObjectMeta(obj interface{}) (metav1.ObjectMeta, error) {
-	cluster, err := key.ToCluster(obj)
-	if err != nil {
-		return metav1.ObjectMeta{}, microerror.Mask(err)
-	}
+func (r *Resource) Name() string {
+	return "alert"
+}
 
-	return metav1.ObjectMeta{
-		Name:      "prometheus-meta-operator",
-		Namespace: key.Namespace(cluster),
-		Labels: map[string]string{
-			key.ClusterIDKey(): key.ClusterID(cluster),
-		},
-	}, nil
+type TemplateData struct {
+	ClusterID    string
+	Installation string
+	ManagedBy    string
+	Namespace    string
 }
 
-func getRules(obj interface{}, installation string) (metav1.Object, error) {
-	objectMeta, err := getObjectMeta(obj)
+func getRules(obj interface{}, installation string) ([]*promv1.PrometheusRule, error) {
+	cluster, err := key.ToCluster(obj)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	cluster, err := key.ToCluster(obj)
+	var data TemplateData = TemplateData{
+		ClusterID:    key.ClusterID(cluster),
+		Installation: installation,
+		ManagedBy:    project.Name(),
+		Namespace:    key.Namespace(cluster),
+	}
+
+	template, err := template.RenderTemplate(data, rulesFileLocation)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	r := &promv1.PrometheusRule{
-		ObjectMeta: objectMeta,
-		Spec: promv1.PrometheusRuleSpec{
-			Groups: []promv1.RuleGroup{
-				rules.LabellingSchemaValidationRule(cluster, installation),
-				rules.Heartbeat(cluster, installation),
-			},
-		},
+	// Split multi yaml files
+	files := bytes.Split(template, []byte("---"))
+
+	var rules []*promv1.PrometheusRule = make([]*promv1.PrometheusRule, 0)
+	for _, file := range files {
+		if len(bytes.TrimSpace(file)) > 0 {
+			var rule promv1.PrometheusRule = promv1.PrometheusRule{}
+
+			if err = yaml.Unmarshal(file, &rule); err != nil {
+				return nil, microerror.Mask(err)
+			}
```

Comment:
Given that this might fail when we provide invalid rule files, we should add unit test in order to validate the rule yaml template we are adding. But this can be done in a separate PR.

Reply:
Yes we should :D